plugins/batterystat.koplugin/main.lua

```diff
@@ -207,10 +202,47 @@ function BatteryStat:showStatistics()
     table.insert(kv_pairs, "----------")
     table.insert(kv_pairs, {_("Historical records are dumped to"), ""})
     table.insert(kv_pairs, {self.dump_file, ""})
-    UIManager:show(KeyValuePage:new{
+    table.insert(kv_pairs, "----------")
+    table.insert(kv_pairs, {_("Should you like to reset the data,"), "",
+                            callback = function()
+                                self:resetAll()
+                                self:restart()
+                            end})
+    table.insert(kv_pairs, {_("please tap here."), "",
+                            callback = function()
+                                self:resetAll()
+                                self:restart()
+                            end})
+    self.kv_page = KeyValuePage:new{
         title = _("Battery statistics"),
         kv_pairs = kv_pairs,
-    })
+    }
+    UIManager:show(self.kv_page)
+end
+
+function BatteryStat:reset(withCharging, withDischarging)
+    self:dumpToText()
+    self.awake = Usage:new()
+    self.sleeping = Usage:new()
+
+    if withCharging then
+        self.charging = Usage:new()
+    end
+    if withDischarging then
+        self.discharging = Usage:new()
+    end
+end
+
+function BatteryStat:resetAll()
+    self:reset(true, true)
+    self.charging_state = State:new()
+    self.awake_state = State:new()
+end
+
+function BatteryStat:restart()
+    assert(self.kv_page ~= nil)
```

Comment:
It's indeed a culture of Google :). And in MS, because of the heritage from client software development, assert is usually disabled in production. But I still find the code quality in Google is much higher than in MS.
Personally, I really like this culture, e.g. no error is acceptable, even it's recoverable.
An unexpected situation means something is wrong: it may be recoverable, but the side effect is a more serious result.

Maybe I can add a dassert() function which triggers only when dbg is enabled.